### PR TITLE
trie: disable fnv64a hashing of hashes for bigcache

### PR DESCRIPTION
**Pull request by holiman**
This is a tiny optimization that replaces the default [`fnv64a`](https://github.com/allegro/bigcache/blob/master/fnv.go) hasher with a custom hasher. The idea being, that since our keys are already hashes, we can use those directly instead of hashing it one extra time.

This will panic if a key shorter than `8` bytes is used, but at least currently, all `Set` operations are on the form `db.cleans.Set(string(hash[:]), enc)` where `hash` is a `common.Hash`, which is always `32` bytes.

### CODE REVIEW

File: trie/database.go [hunk 1]
```diff
@@ -257,6 +258,17 @@ func expandNode(hash hashNode, n node) node {
 	}
 }
 
+// trienodeHasher is a struct to be used with BigCache, which uses a Hasher to determine which shard to place an
+// entry into. It's not a cryptographic hash, just to provide a bit of anti-collision.
+// By default, FNV64a is used
+// Since trie keys are already hashes, we can just use the key directly to map shard id
```
Comment (karalabe) [inline trie/database.go]: Nitpick, but could you please justify this (and your other comments in other PRs ;P) to +-80 characters. It looks so nasty that the comment is totally randomly jagged.

Comment (karalabe) [reply trie/database.go]: Also, please use full sentences for docs (i.e. period at the end).

Comment (holiman) [reply trie/database.go]: fixed

File: trie/database.go [hunk 2]
```diff
@@ -257,6 +258,19 @@ func expandNode(hash hashNode, n node) node {
...
 
+// trienodeHasher is a struct to be used with BigCache, which uses a Hasher to
+// determine which shard to place an entry into. It's not a cryptographic hash,
+// just to provide a bit of anti-collision (default is FNV64a).
+//
+// Since trie keys are already hashes, we can just use the key directly to
+// map shard id.
+type trienodeHasher struct{}
+
+// Sum64 implements the bigcache.Hasher interface
```
Comment (karalabe) [inline trie/database.go]: . ;P

Comment (holiman) [reply trie/database.go]: ah damn you, fixed

**Comment by holiman** [comment]
Well, we could benchmark it for fun, but I'm not sure it's worth it.. It should do one full sync before release, but I'm thinking it's probably ok to merge to master .. ?

**Comment by karalabe** [comment]
YOLO it is!

**Pull request merged**